compiler/lib/generate.ml

```diff
@@ -501,55 +501,165 @@ let constant ~ctx x level =
 type queue_elt =
   { prop : int
   ; ce : J.expression
-  ; loc : J.location
```

Comment:
The `let*` rewrite make it hard to review the rest of the diff. could this commit be split, maybe ?

Reply:
It is splitted.